programs/mmm/src/instructions/cnft/sol_cnft_fulfill_buy.rs

```diff
@@ -134,7 +134,7 @@ pub fn handler<'info>(
 ) -> Result<()> {
     // let payer = &ctx.accounts.payer;
     let owner = &ctx.accounts.owner;
-    let pool = &mut ctx.accounts.pool;
+    let pool = &ctx.accounts.pool;
```

Comment:
this needs to be immutable if you're going to transfer ownership to the pool.
based off the other fulfill_buy ixs the owner should be either the buyer or the pool depending on if you're reinvesting. left the test as buyer for now

Reply:
good learning!